### ISSUE TITLE
Getting "asyncio - Unclosed connector" when calling blob_storage_list

### ISSUE THREAD

**Issue by diego-infostrux** [PrefectHQ/prefect#16057]
### Bug summary

When trying to list content from an Azure Blob storage, I get an error message saying:

```
09:09:57.794 | ERROR   | asyncio - Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x0000021FFECF5A60>
09:09:57.794 | ERROR   | asyncio - Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x0000021FFECEB170>, 2615240.078)]']
connector: <aiohttp.connector.TCPConnector object at 0x0000021FFECF5BB0>
```

### Version info

```Text
Version:             3.1.2
API version:         0.8.4
Python version:      3.12.2
Git commit:          02b99f0a
Built:               Tue, Nov 12, 2024 1:38 PM
OS/Arch:             win32/AMD64
Profile:             local
Server type:         cloud
Pydantic version:    2.9.2
Integrations:
  prefect-azure:     0.4.1
  prefect-dbt:       0.6.3
  prefect-snowflake: 0.28.0
```

### Additional context

```python
from prefect import flow
from prefect_azure import AzureBlobStorageCredentials
from prefect_azure.blob_storage import blob_storage_list
import os

@flow
def example_blob_storage_list_flow():
    blob_storage_credentials = AzureBlobStorageCredentials(account_url=os.getenv("AZURE_CONTAINER_URL"))

    data = blob_storage_list(
        container=os.getenv("AZURE_CONTAINER_NAME"),
        blob_storage_credentials=blob_storage_credentials
    )
    return data

if __name__ == "__main__":
    result = example_blob_storage_list_flow()    
    for i in result:
        print(i['name'])`
```

**Comment by diego-infostrux** [comment]
aiohttp==3.10.10

**Comment by desertaxle** [comment]
Thanks for the issue @diego-infostrux! 

`blob_storage_list` is an `async` task, so I think this error might be caused by the call to `blob_storage_list` not being awaited.

Can you try this and see if it resolves the issue?
```python
from prefect import flow
from prefect_azure import AzureBlobStorageCredentials
from prefect_azure.blob_storage import blob_storage_list
import asyncio
import os

@flow
async def example_blob_storage_list_flow():
    blob_storage_credentials = AzureBlobStorageCredentials(account_url=os.getenv("AZURE_CONTAINER_URL"))

    data = await blob_storage_list(
        container=os.getenv("AZURE_CONTAINER_NAME"),
        blob_storage_credentials=blob_storage_credentials
    )
    return data

if __name__ == "__main__":
    result = asyncio.run(example_blob_storage_list_flow())  
    for i in result:
        print(i['name'])`
```

**Comment by diego-infostrux** [comment]
Still get same error:
![image](https://github.com/user-attachments/assets/baf95f24-ffeb-4c4d-9180-c3bdceb0e4e0)
![image](https://github.com/user-attachments/assets/f9e07aa6-a859-4233-a30a-b241372778b7)

**Comment by diego-infostrux** [comment]
$ pip freeze
WARNING: Ignoring invalid distribution ~refect (c:\Users\djuritz\workspace\prefect\.venv\Lib\site-packages)
agate==1.9.1
aiohappyeyeballs==2.4.3
aiohttp==3.11.5
aiosignal==1.3.1
aiosqlite==0.20.0
alembic==1.13.3
annotated-types==0.7.0
anyio==4.6.2.post1
apprise==1.9.0
asgi-lifespan==2.1.0
asgiref==3.8.1
asn1crypto==1.5.1
asyncpg==0.30.0
attrs==24.2.0
azure-common==1.1.28
azure-core==1.31.0
azure-identity==1.19.0
azure-mgmt-containerinstance==10.1.0
azure-mgmt-core==1.4.0
azure-mgmt-resource==23.2.0
azure-storage-blob==12.23.1
babel==2.16.0
backoff==2.2.1
cachetools==5.5.0
certifi==2024.8.30
cffi==1.17.1
charset-normalizer==3.4.0
click==8.1.7
cloudpickle==3.1.0
colorama==0.4.6
coolname==2.2.0
croniter==3.0.4
cryptography==43.0.3
daff==1.3.46
dateparser==1.2.0
dbt-adapters==1.7.0
dbt-common==1.10.0
dbt-core==1.8.8
dbt-extractor==0.5.1
dbt-semantic-interfaces==0.5.1
deepdiff==7.0.1
Deprecated==1.2.14
docker==7.1.0
exceptiongroup==1.2.2
fastapi==0.115.4
filelock==3.16.1
frozenlist==1.5.0
fsspec==2024.10.0
googleapis-common-protos==1.65.0
graphql-core==3.2.5
graphviz==0.20.3
greenlet==3.1.1
griffe==1.5.1
grpcio==1.67.1
h11==0.14.0
h2==4.1.0
hpack==4.0.0
httpcore==1.0.6
httpx==0.27.2
humanize==4.11.0
hyperframe==6.0.1
idna==3.10
importlib-metadata==6.11.0
isodate==0.6.1
Jinja2==3.1.4
jinja2-humanize-extension==0.4.0
jsonpatch==1.33
jsonpointer==3.0.0
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
leather==0.4.0
Logbook==1.5.3
Mako==1.3.6
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
mashumaro==3.14
mdurl==0.1.2
minimal-snowplow-tracker==0.0.2
more-itertools==10.5.0
msal==1.31.0
msal-extensions==1.2.0
msgpack==1.1.0
multidict==6.1.0
networkx==3.4.2
oauthlib==3.2.2
opentelemetry-api==1.28.1
opentelemetry-exporter-otlp==1.27.0
opentelemetry-exporter-otlp-proto-common==1.27.0
opentelemetry-exporter-otlp-proto-grpc==1.13.0
opentelemetry-exporter-otlp-proto-http==1.13.0
opentelemetry-instrumentation==0.34b0
opentelemetry-instrumentation-asgi==0.34b0
opentelemetry-instrumentation-asyncpg==0.34b0
opentelemetry-instrumentation-fastapi==0.34b0
opentelemetry-instrumentation-httpx==0.34b0
opentelemetry-instrumentation-logging==0.48b0
opentelemetry-instrumentation-sqlalchemy==0.34b0
opentelemetry-proto==1.13.0
opentelemetry-sdk==1.13.0
opentelemetry-semantic-conventions==0.34b0
opentelemetry-util-http==0.34b0
ordered-set==4.1.0
orjson==3.10.10
packaging==24.1
parsedatetime==2.6
pathspec==0.12.1
pendulum==3.0.0
platformdirs==4.3.6
portalocker==2.10.1
prefect==3.1.2
prefect-azure==0.4.1
prefect-dbt==0.6.3
prefect-opentelemetry==0.1.0
prefect-snowflake==0.28.0
prometheus_client==0.21.0
propcache==0.2.0
protobuf==3.20.3
pycparser==2.22
pydantic==2.9.2
pydantic-extra-types==2.9.0
pydantic-settings==2.6.0
pydantic_core==2.23.4
Pygments==2.18.0
PyJWT==2.9.0
pyOpenSSL==24.2.1
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-slugify==8.0.4
pytimeparse==1.1.8
pytz==2024.2
pywin32==308
PyYAML==6.0.2
readchar==4.2.0
referencing==0.35.1
regex==2024.9.11
requests==2.32.3
requests-oauthlib==2.0.0
rfc3339-validator==0.1.4
rich==13.9.3
rpds-py==0.20.0
ruamel.yaml==0.18.6
ruamel.yaml.clib==0.2.12
setuptools==75.3.0
sgqlc==16.4
shellingham==1.5.4
six==1.16.0
sniffio==1.3.1
snowflake-connector-python==3.12.3
sortedcontainers==2.4.0
SQLAlchemy==2.0.36
sqlparse==0.5.1
starlette==0.41.2
text-unidecode==1.3
time-machine==2.16.0
toml==0.10.2
tomlkit==0.13.2
typer==0.12.5
typing_extensions==4.12.2
tzdata==2024.2
tzlocal==5.2
ujson==5.10.0
urllib3==2.2.3
uvicorn==0.32.0
websockets==13.1
wrapt==1.16.0
yarl==1.17.1
zipp==3.20.2

**Comment by desertaxle** [comment]
I think we are missing an async context manager in the `blob_storage_list` task for the default credentials. I will try a fix and open a PR if it works.

**Comment by desertaxle** [comment]
@diego-infostrux I was right that we were missing a clean-up step when using default Azure credentials. I have a fix in #16071 which we should be able to release today!

**Comment by desertaxle** [comment]
A fix for this issue is available in `prefect-azure==0.4.2`

**Comment by diego-infostrux** [comment]
> A fix for this issue is available in `prefect-azure==0.4.2`

Thanks, it works fine now.
